### PR TITLE
INT-37 (Filter by Status on Click)

### PR DESCRIPTION
**Pull request by cchanyu**
#Notes: 
- Added new svg icons that represents better idea
- Added total count option to revert back to original data
- Changed icons to button, so they are clickable and when they're clicked, it filters according to its status

### CODE REVIEW

File: src/components/Dropdown/StatusChecker.js [hunk 1]
```diff
@@ -1,29 +1,70 @@
...
+    <button type="submit" className="status__box completed" onClick={() => SortBy("Completed", props)}>
       <div className="status__count">{props.completed}</div>
-      <img className="status__icon" src={List} alt="List" />
-    </div>
+      <img className="status__icon" src={Check} alt="Check" />
+    </button>
   </div>
 );
 
+const SortBy = (event, props) => {
```
Comment (horseeyephil) [inline src/components/Dropdown/StatusChecker.js]: lower case function names please!

File: src/App.js [hunk 2]
```diff
@@ -32,11 +33,11 @@ class App extends Component {
       CalcStatus(currDate, endDate, isCompleted);
       return status;
     });
-
     const incompleteCount = updateStatus.filter((project) => project === 'incomplete');
     const inprogressCount = updateStatus.filter((project) => project === 'inprogress');
     const completeCount = updateStatus.filter((project) => project === 'complete');
     this.setState({
```
Comment (horseeyephil) [inline src/App.js]: sorry - I should have caught this in an earlier PR!
so - wondering why this is happening in `componentDidMount` - these React lifecycle hooks are typically used for asynchronous side effects, i-o, etc.
These seem like they can all be calculated on initialization and could go in the constructor

File: src/components/Dropdown/StatusChecker.js [hunk 3]
```diff
@@ -1,29 +1,70 @@
...
+      break;
+
+    default:
+      props.updateDataState(props.originalData);
+      break;
+  }
+}
+
+const StatusIncomplete = (props) => {
+  const cloneData = props.originalData.slice();
```
Comment (horseeyephil) [inline src/components/Dropdown/StatusChecker.js]: cloning will not be necessary! as `.filter` works immutably to create a new array

File: src/components/Dropdown/StatusChecker.js [hunk 4]
```diff
@@ -1,29 +1,70 @@
...
-    </div>
+      <img className="status__icon" src={Check} alt="Check" />
+    </button>
   </div>
 );
 
+const SortBy = (event, props) => {
+  switch (event) {
+    case 'Incomplete':
+      props.updateDataState(StatusIncomplete(props));
```
Comment (horseeyephil) [inline src/components/Dropdown/StatusChecker.js]: I'm glad that `props.updateDataState` is decoupled from the payload `StatusIncomplete()`
However- I'm wondering - this seems to be a lot of `switching` and code for logic that is pretty similar.. Instead of the three individualized functions below - is it possible to use one filtering function?
In fact - it seems like you already have three filtered arrays at the top level, could you just switch the display between them? -no need to filter once more?

File: src/components/Dropdown/StatusChecker.js [hunk 5]
```diff
@@ -1,29 +1,48 @@
...
+    </button>
+    <button type="submit" className="status__box completed" onClick={() => props.updateDataState(StatusCompleted(props))}>
+      <div className="status__count">{StatusCompleted(props).length}</div>
+      <img className="status__icon" src={Check} alt="Check" />
+    </button>
   </div>
 );
 
+
+const StatusIncomplete = (props) => {
```
Comment (richsong) [inline src/components/Dropdown/StatusChecker.js]: lower case function names please

File: src/components/Dropdown/StatusChecker.js [hunk 6]
```diff
@@ -1,29 +1,48 @@
...
+  const currDate = moment().format('L');
+  const modifiedData = props.originalData.filter((project) => (project.isCompleted === false && moment(currDate).isBefore(moment(project.endDate))));
+  return modifiedData;
+};
+
+const StatusCompleted = (props) => {
+  const modifiedData = props.originalData.filter((project) => project.isCompleted === true);
+  return modifiedData;
+};
+
```
Comment (richsong) [inline src/components/Dropdown/StatusChecker.js]: we are missing props validation?

File: src/App.js [hunk 7]
```diff
@@ -12,46 +12,24 @@ class App extends Component {
   constructor(props) {
     super(props);
     this.data = {
-      originalData: data.projects,
+      originalData: data.projects
```
Comment (richsong) [inline src/App.js]: fix linting issues

File: src/components/Dropdown/StatusChecker.js [hunk 8]
```diff
@@ -1,29 +1,48 @@
...
-      <img className="status__icon" src={Check} alt="Check" />
-    </div>
-    <div className="status__box completed">
-      <div className="status__count">{props.completed}</div>
+    <button type="submit" className="status__box totalCount" onClick={() => props.updateDataState(props.originalData)} >
+      <div className="status__count">{props.originalData.length}</div>
       <img className="status__icon" src={List} alt="List" />
-    </div>
+    </button>
+    <button type="submit" className="status__box incomplete" onClick={() => props.updateDataState(StatusIncomplete(props))}>
```
Comment (richsong) [inline src/components/Dropdown/StatusChecker.js]: `Block__element--modifier`
We should be naming our classes like so under the BEM specification:
http://getbem.com/naming/

File: src/components/Dropdown/StatusChecker.js (hunk 5 above)
Comment (richsong) [reply src/components/Dropdown/StatusChecker.js]: move the function declarations above the component

File: src/components/Dropdown/StatusChecker.scss [hunk 9]
```diff
@@ -38,4 +38,8 @@ $background: #cccccc;
 
 .completed {
     background-color: green;
+}
+
+.totalCount {
+    background-color: black;
```
Comment (richsong) [inline src/components/Dropdown/StatusChecker.scss]: this is not part of BEM. Please see above comment. It should be nested within `.status {}`

File: src/components/Dropdown/StatusChecker.js [hunk 10]
```diff
@@ -1,29 +1,60 @@
...
+      </button>
     </div>
-  </div>
-);
+  );
+};
+
+StatusChecker.propTypes = {
+  updateDataState: PropTypes.func.isRequired,
+  originalData: PropTypes.instanceOf(Array).isRequired,
```
Comment (richsong) [inline src/components/Dropdown/StatusChecker.js]: Ok the right way of doing this requires a combination of two things:
`PropType..arrayOf()` and `PropTypes.shape({})`

Take a quick look at those things and read it over.

Let's try to do a better job with prop types validation. We want to start enforcing more type checking as it will become more valuable as a tool while the application grows at scale. Why? Because runtime assertions and type checking will help clear up any easily avoidable issues going forward.

So... it should look something like:
```
originalData: PropTypes.arrayOf(
    PropType.shapeOf({
        ...restOfPropertiesFromProjectsObject
        })
    )
```

Basically we are saying that originalData is an array of OBJECTS. The object is defined by `.shapeOf` and it has these properties. Let me know if you have any questions
